File: src/main/java/com/example/tigregym/entities/Matricula.java
package com.example.tigregym.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Matricula {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public long id;

    public String aluno;

    public String plano;

    public LocalDateTime dataInicio;

    public LocalDateTime dataVencimento;

    public String status;

}

File: front/app/(sistema)/matricula/matricula.ts
export class Matricula {
  constructor(
    public id: number | null,
    public aluno: string,
    public plano: string,
    public dataInicio: string,
    public dataVencimento: string,
    public status: string
  ) {}
}

export interface MatriculaFormProps {
  matriculaExistente?: Matricula;
}
